repository/kafka/docs/latest/custom.md

```diff
@@ -0,0 +1,198 @@
+
+Example custom configuration:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: custom-configuration
+data:
+  server.properties: |
+    ssl.secure.random.implementation=SHA1PRNG
+    connection.failed.authentication.delay.ms=300
+```
+
+Create the ConfigMap in the namespace we will have the KUDO Kafka cluster
+
+```
+> kubectl create -f custom-configuration.yaml -n kudo-kafka
+configmap/custom-configuration created
+```
+
+Verify the ConfigMap is created correctly 
+
+```
+> kubectl get configmap custom-configuration -n kudo-kafka -o yaml
+apiVersion: v1
+data:
+  server.properties: |
+    ssl.secure.random.implementation=SHA1PRNG
+    connection.failed.authentication.delay.ms=300
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2019-10-19T12:41:19Z"
+  name: custom-configuration
+  namespace: kudo-kafka
+  resourceVersion: "349819"
+  selfLink: /api/v1/namespaces/kudo-kafka/configmaps/custom-configuration
+  uid: 8a9f1520-ed3d-4b90-b544-a3e20d20fa5f
+```
+
+Now we are ready to start the KUDO Kafka cluster with custom configuration to be used with default tuned configuration. 
+
+```
+kubectl kudo install kafka \
+    --instance=kafka --namespace=kudo-kafka \
+    -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
+    -p BROKER_COUNT=3 \
+    -p CUSTOM_SERVER_PROPERTIES_CM_NAME=custom-configuration 
+```
+
+Verify in the logs if the custom configuration is being used in the Apache Kafka brokers and the `KafkaConfig` is correctly using `connection.failed.authentication.delay.ms` with value of `300` 
+
+```
+> kubectl logs kafka-kafka-0 -n kudo-kafka
+[2019-10-21 13:46:58,325] Appending custom configuration file to the server.properties...
+ssl.secure.random.implementation=SHA1PRNG
+connection.failed.authentication.delay.ms=300
+Starting the kafka broker using broker.id 0...
+[ ... lines removed for clarity ...]
+	compression.type = producer
+	connection.failed.authentication.delay.ms = 300
+	connections.max.idle.ms = 600000
+	connections.max.reauth.ms = 0
+[ ... lines removed for clarity ...]
+```
+#### Updating the custom configuration
+
+The KUDO Kafka custom configuration ConfigMap isn't watched by the KUDO controller. Therefore any updates done in the custom configuration will need a later rolling restart.
+
+Edit the configmap with changes we want to rollout:
+
+```
+> kubectl edit configmap custom-configuration -n kudo-kafka
+configmap/custom-configuration edited
+```
+
+Perform a rolling restart on the statefulset to reload the configmap.
+
+```
+> kubectl rollout restart statefulset kafka-kafka -n kudo-kafka
+statefulset.apps/kafka-kafka restarted
+```
+
+#### :warning: Excludelist of the custom configuration
+
+Users can update all the broker configuration properties except the `broker.id`, `listeners`,`advertised.listeners`,  `advertised.host.name`,  `listener.security.protocol.map` `log.dirs`
```

Comment:
Let's make each of these their own bullet to emphasize this list cannot be edited. Did you add these to the `not-allowed` plan (or maybe they already were?)

Reply:
a warning is printed any of those are used. we cannot use any plan as these aren't parameters but part of custom configuration
example: `WARN: cannot override the log.dirs using custom properties configmap.`

Reply:
added bullets for each property